packages/@romejs/compiler/transforms/compile/transpile/enums.ts

```diff
@@ -0,0 +1,248 @@
+import {
+	AnyJSExpression,
+	AnyJSStatement,
+	JSBinaryExpression,
+	JSBindingIdentifier,
+	JSCallExpression,
+	JSFunctionExpression,
+	JSUnaryExpression,
+	TSEnumDeclaration,
+	jsNumericLiteral,
+	jsStringLiteral,
+} from "@romejs/ast";
+import {LetBinding, Path, VarBinding} from "@romejs/compiler";
+import {REDUCE_REMOVE} from "@romejs/compiler/constants";
+import {template} from "@romejs/js-ast-utils";
+
+interface PreviousEnumMembers {
+	[name: string]: number | string;
+}
+
+function buildEnumWrapper(
+	id: JSBindingIdentifier,
+	assignments: Array<AnyJSExpression>,
+): AnyJSExpression {
+	const expressionStatement = (template.expression`
+		(function (${id}) {})(${id} || (${id} = {}));
+	` as JSCallExpression);
+	const functionExpression = (expressionStatement.callee as JSFunctionExpression);
+	functionExpression.body.body = (assignments as Array<AnyJSStatement>);
+	return (expressionStatement as AnyJSExpression);
+}
+
+function buildEnumMember(
+	isString: boolean,
+	id: JSBindingIdentifier,
+	name: string,
+	value: AnyJSExpression,
+): AnyJSExpression {
+	return (isString ? buildStringAssignment : buildNumericAssignment)(
+		id,
+		name,
+		value,
+	);
```

Comment:
I think this is unnecessary _complex_.
Could you change it this way?

```suggestion
return isString
  ? buildStringAssignment(id, name, value)
  : buildNumericAssignment(id, name, value);
```